auth/src/main/java/org/aerogear/auth/IUserPrincipal.java

```diff
@@ -11,11 +11,19 @@
 public interface IUserPrincipal extends Principal {
 
     /**
-     * Checks if the user has the specified role.
+     * Checks if the user has the specified Client role.
      * @param role role to be checked
+     * @param clientId clientID related to role
      * @return true or false
      */
-    boolean hasRole(IRole role);
+    boolean hasClientRole(String role, String clientId);
```

Comment:
I prefer to put the `clientId` in front as it is the name space for the role.

Reply:
actually, I am question if we need the `clientId` here at all as the value of `clientId` is already provided in the configuration file for the authentication service?